naga/src/front/spv/mod.rs

```diff
@@ -560,25 +561,40 @@ struct BlockContext<'function> {
     parameter_sampling: &'function mut [image::SamplingFlags],
 }
 
+impl<'a> BlockContext<'a> {
+    /// Recurse into the expression with the given handle, locating any contained
+    /// global variables.
+    ///
+    /// This is used to track atomic upgrades.
+    fn get_contained_global_variables(
+        &self,
+        handle: Handle<crate::Expression>,
+    ) -> IndexSet<Handle<crate::GlobalVariable>> {
+        let mut s = IndexSet::default();
+        if let Ok(expr) = self.expressions.try_get(handle) {
+            #[allow(clippy::pattern_type_mismatch)]
```

Comment:
I know it's weird, but please go ahead and use `*expr`. It's Naga style. Maybe it will change in the future, but it's not going to change in this PR.